metrics/ci_integrations/test/destination/firestore/config/validation_delegate/firestore_destination_validation_delegate_test.dart

```diff
@@ -0,0 +1,360 @@
+
+        expect(result.isFailure, isTrue);
+      },
+    );
+
+    test(
+      ".validatePublicApiKey() returns a result with the 'api key invalid' additional context, if the auth throws a firebase auth exception with the 'invalid api key' code",
+      () async {
+        whenSignIn().thenAnswer(
+          (_) => Future.error(invalidApiKeyAuthException),
+        );
+
+        final result = await delegate.validatePublicApiKey(apiKey);
+
+        expect(
+          result.additionalContext,
+          equals(FirestoreStrings.apiKeyInvalid),
+        );
+      },
+    );
+
+    test(
+      ".validatePublicApiKey() returns a successful field validation result, if the auth throws a firebase auth exception with the exception code that does not indicate the invalid api key",
+      () async {
+        final exceptionCodes = FirebaseAuthExceptionCode.values.where(
+          (code) => code != FirebaseAuthExceptionCode.invalidApiKey,
+        );
+
+        for (final exceptionCode in exceptionCodes) {
+          final exception = FirebaseAuthException(exceptionCode, message);
+          whenSignIn().thenAnswer((_) => Future.error(exception));
+
+          final result = await delegate.validatePublicApiKey(apiKey);
+
+          expect(result.isSuccess, isTrue);
+        }
+      },
+    );
+
+    test(
+      ".validatePublicApiKey() returns a successful field validation result if the public api key is valid",
+      () async {
+        whenSignIn().thenAnswer((_) => Future.value(user));
+
+        final result = await delegate.validatePublicApiKey(apiKey);
+
+        expect(result.isSuccess, isTrue);
+      },
+    );
+
+    test(
+      ".validateAuth() creates a firebase auth with the firebase auth factory",
+      () {
+        whenCreateFirebaseAuth().thenReturn(firebaseAuth);
+
+        delegate.validateAuth(credentials);
+
+        verify(authFactory.create(apiKey)).called(once);
+      },
+    );
+
+    test(
+      ".validateAuth() signs in with the created firebase auth and the given credentials",
+      () {
+        final expectedEmail = credentials.email;
+        final expectedPassword = credentials.password;
+        whenCreateFirebaseAuth().thenReturn(firebaseAuth);
+
+        delegate.validateAuth(credentials);
+
+        verify(
+          firebaseAuth.signIn(expectedEmail, expectedPassword),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".validateAuth() returns a failure field validation result, if the auth throws a firebase auth exception with the 'email not found' code",
+      () async {
+        whenSignIn().thenAnswer(
+          (_) => Future.error(emailNotFoundAuthException),
+        );
+
+        final result = await delegate.validateAuth(credentials);
+
+        expect(result.isFailure, isTrue);
+      },
+    );
+
+    test(
+      ".validateAuth() returns a field validation result with the exception message if the auth throws a firebase auth exception with the 'email not found' code",
```

Comment:
```suggestion
      ".validateAuth() returns a field validation result containing the exception message if the auth throws a firebase auth exception with the 'email not found' code",
```
Would it be a bit more clear?